sight_service/worklist_scheduler_opt.py

```diff
@@ -187,71 +166,64 @@ def decision_point(
-        sample = self.active_samples[request.worker_id]['sample']
-      else:
-        raise ValueError("key not foung in active_samples")
-    next_action = sample[0]
-    logging.info('next_action=%s', next_action)
-    # raise SystemExit
-    dp_response.action.extend(param_dict_to_proto(next_action))
-    # print('self.active_samples : ', self.active_samples)
-    # print('self.pending_samples : ', self.pending_samples)
-    # print('self.completed_samples : ', self.completed_samples)
-    dp_response.action_type = service_pb2.DecisionPointResponse.ActionType.AT_ACT
-    # else:
-    #   dp_response.action_type = service_pb2.DecisionPointResponse.ActionType.AT_RETRY
-
-    logging.debug("<<<<  Out %s of %s", method_name, _file_name)
-    return dp_response
+    # with self.active_lock.gen_rlock():
+    #     if (request.worker_id in self.active_samples):
+    #         sample = self.active_samples[request.worker_id]['sample']
+    #     else:
+    #         raise ValueError("key not foung in active_samples")
+    # next_action = sample[0]
+    # logging.info('next_action=%s', next_action)
+    # # raise SystemExit
+    # dp_response.action.extend(param_dict_to_proto(next_action))
+    # dp_response.action_type = service_pb2.DecisionPointResponse.ActionType.AT_ACT
+    # logging.debug("<<<<  Out %s of %s", method_name, _file_name)
+    # return dp_response
 
   @overrides
   def finalize_episode(
       self, request: service_pb2.FinalizeEpisodeRequest
-  ) -> service_pb2.FinalizeEpisodeResponse:
+    ) -> service_pb2.FinalizeEpisodeResponse:
     method_name = "finalize_episode"
     logging.debug(">>>>  In %s of %s", method_name, _file_name)
 
-    # logging.info("req in finalize episode of dummy.py : %s", request)
-
-    with self.active_lock.gen_rlock():
-      sample_dict = self.active_samples[request.worker_id]
-
-    with self.completed_lock.gen_wlock():
-      self.completed_samples[sample_dict['id']] = {
-          # 'action': self.pending_samples[unique_action_id],
-          'action':
-              param_proto_to_dict(request.decision_point.choice_params),
-          'attribute':
-              sample_dict['sample'][1],
-          'reward':
-              request.decision_outcome.reward,
-          'outcome':
-              param_proto_to_dict(request.decision_outcome.outcome_params)
-      }
-
-    with self.active_lock.gen_wlock():
-      del self.active_samples[request.worker_id]
-
-    # print('self.active_samples : ', self.active_samples)
-    # print('self.pending_samples : ', self.pending_samples)
-    # print('self.completed_samples : ', self.completed_samples)
+    logging.info("self.queue => %s", self.queue)
+
+    all_messages = self.queue.get_all_messages()
+
+    active_messages = all_messages["active"][request.worker_id]
+
+    for action_id, message in active_messages.items():
+      self.queue.complete_message(
+          message_id=action_id,
+          worker_id=request.worker_id,
+          extra_details={
```

Comment:
Can this dict be a dataclass that provides a method to add itself to a service_pb2.GetOutcomeResponse() proto? That would pull this logic into a single place.

Reply:
I didn't exactly get this , could you elaborate

Reply:
what I had in mind was:
@dataclass
class MessageDetails:
              action: Dict
              attribute: Dict
              reward: float
              outcome: Dict
  @abstractmethod
  def create(cls, attributes, req: FinalizeEpisodeRequest):
     return MessageDetails(...)
     
  def add_to_outcome_response(self, sample_id: int, response: service_pb2.GetOutcomeResponse):
        outcome = response.outcome.add()
        outcome.action_id = sample_id
        outcome.status = service_pb2.GetOutcomeResponse.Outcome.Status.COMPLETED
        outcome.reward = self.reward
        outcome.action_atrrs.extend(self.action)
        outcome.outcome_attrs.extend(self.outcome)
        outcome.attributes.extend(self.attribute)

Reply:
updating the msg with data class object